diff --git a/build/config/clang/BUILD.gn b/build/config/clang/BUILD.gn
index d49f0ec758..1cc64a3a85 100644
--- a/build/config/clang/BUILD.gn
+++ b/build/config/clang/BUILD.gn
@@ -177,7 +177,21 @@ template("clang_lib") {
         }
       } else if (is_apple) {
         _dir = "darwin"
-      } else if (is_linux || is_chromeos) {
+      } else if (is_linux) {
+        if (current_cpu == "x64") {
+          _dir = "x86_64-redhat-linux-gnu"
+        } else if (current_cpu == "x86") {
+          _dir = "i386-redhat-linux-gnu"
+        } else if (current_cpu == "arm64") {
+          _dir = "aarch64-redhat-linux-gnu"
+        } else if (current_cpu == "arm") {
+          _dir = "armhf-redhat-linux-gnu"
+        } else if (current_cpu == "ppc64") {
+          _dir = "ppc64le-redhat-linux-gnu"
+        } else {
+          assert(false)  # Unhandled cpu type
+        }
+      } else if (is_chromeos) {
         if (current_cpu == "x64") {
           _dir = "x86_64-unknown-linux-gnu"
         } else if (current_cpu == "x86") {
diff --git a/build/rust/rust_bindgen.gni b/build/rust/rust_bindgen.gni
index 05a660ddfc..0e9a14fb18 100644
--- a/build/rust/rust_bindgen.gni
+++ b/build/rust/rust_bindgen.gni
@@ -19,14 +19,14 @@ if (host_os == "win") {

 # On Windows, the libclang.dll is beside the bindgen.exe, otherwise it is in
 # ../lib.
-_libclang_path = rust_bindgen_root
+_libclang_path = clang_base_path
 if (!use_chromium_rust_toolchain &&
     (current_cpu == "ppc64" || current_cpu == "s390x")) {
   _libclang_path = rust_sysroot_absolute + "/lib64"
 } else if (host_os == "win") {
   _libclang_path += "/bin"
 } else {
-  _libclang_path += "/lib"
+  _libclang_path += "/lib64"
 }

 # Template to build Rust/C bindings with bindgen.
diff --git a/build/rust/rust_bindgen_generator.gni b/build/rust/rust_bindgen_generator.gni
index 0ddb769276..3bca82f6c0 100644
--- a/build/rust/rust_bindgen_generator.gni
+++ b/build/rust/rust_bindgen_generator.gni
@@ -26,14 +26,14 @@ if (!use_chromium_rust_toolchain &&

 # On Windows, the libclang.dll is beside the bindgen.exe, otherwise it is in
 # ../lib.
-_libclang_path = rust_bindgen_root
+_libclang_path = clang_base_path
 if (!use_chromium_rust_toolchain &&
     (host_cpu == "ppc64" || host_cpu == "s390x")) {
   _libclang_path = rust_sysroot_absolute + "/lib64"
 } else if (host_os == "win") {
   _libclang_path += "/bin"
 } else {
-  _libclang_path += "/lib"
+  _libclang_path += "/lib64"
 }

 # Template to build Rust/C bindings with bindgen.
