diff --git a/sandbox/policy/linux/bpf_audio_policy_linux.cc b/sandbox/policy/linux/bpf_audio_policy_linux.cc
index e1716b97e8..3c24c69d02 100644
--- a/sandbox/policy/linux/bpf_audio_policy_linux.cc
+++ b/sandbox/policy/linux/bpf_audio_policy_linux.cc
@@ -60,6 +60,10 @@ ResultExpr AudioProcessPolicy::EvaluateSyscall(int system_call_number) const {
 #if defined(__NR_ioctl)
     case __NR_ioctl:
 #endif
+#if defined(__NR_memfd_create)
+    case __NR_memfd_create:
+      // TODO(crbug.com/442771181): temporary allowance for crasher.
+#endif
 #if defined(__NR_pwrite)
     case __NR_pwrite:
 #endif
