diff --git a/sandbox/policy/linux/bpf_audio_policy_linux.cc b/sandbox/policy/linux/bpf_audio_policy_linux.cc
index e1716b97e8..a4e35e1cb5 100644
--- a/sandbox/policy/linux/bpf_audio_policy_linux.cc
+++ b/sandbox/policy/linux/bpf_audio_policy_linux.cc
@@ -118,6 +118,12 @@ ResultExpr AudioProcessPolicy::EvaluateSyscall(int system_call_number) const {
       const Arg<int> domain(0);
       return If(domain == AF_UNIX, Allow()).Else(Error(EPERM));
     }
+#endif
+#if defined(__NR_memfd_create)
+    case __NR_memfd_create): {
+      const Arg<int> flags(1);
+      return If((flags & ~(MFD_NOEXEC_SEAL), Allow()).Else(CrashSIGSYS());
+    }
 #endif
     default:
 #if defined(__x86_64__)
