diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 59942a3cf6..06ae018422 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -373,14 +373,23 @@ config("compiler") {
       # On Aarch64, SCS requires the x18 register to be unused because it will hold
       # a pointer to the shadow stack. For Android we know that Clang doesn't use
       # x18 by default. On other OSs adding "-ffixed-x18" might be required.
-      assert(is_android)
+      assert(is_android || is_linux)
 
       scs_parameters = [
         "-fsanitize=shadow-call-stack",
-        "-fno-stack-protector",
+        "-fstack-protector-strong",
       ]
+      rs_scs_params = [ "-Zsanitize=shadow-call-stack" ]
+
+      # fixed-x18 may be required for ShadowCallStack
+      if (is_linux) {
+        scs_paramaters += [ "-ffixed-x18" ]
+        rs_scs_params += [ "-Zfixed-x18" ]
+      }
+
       cflags += scs_parameters
       ldflags += scs_parameters
+      rustflags += rs_scs_params
     } else {
       if (is_apple) {
         # The strong variant of the stack protector significantly increases
@@ -400,6 +409,13 @@ config("compiler") {
       }
     }
 
+    if (is_linux) {
+      cflags += [ "-fstack-clash-protection" ]
+
+      cflags += [ "-mharden-sls=all" ]
+      rustflags += [ "-Ctarget-feature=+harden-sls-ijmp", "-Ctarget-feature=+harden-sls-ret" ]
+    }
+
     if (use_lld) {
       ldflags += [ "-fuse-ld=lld" ]
       if (lld_path != "") {
@@ -2282,7 +2298,7 @@ config("chromium_code") {
       # Non-chromium code is not guaranteed to compile cleanly with
       # _FORTIFY_SOURCE. Also, fortified build may fail when optimizations are
       # disabled, so only do that for Release build.
-      fortify_level = "2"
+      fortify_level = "3"
 
       # ChromeOS's toolchain supports a high-quality _FORTIFY_SOURCE=3
       # implementation with a few custom glibc patches. Use that if it's
@@ -3363,8 +3379,14 @@ buildflag_header("compiler_buildflags") {
 }
 
 config("cet_shadow_stack") {
-  if (enable_cet_shadow_stack && is_win) {
+  if (enable_cet_shadow_stack) {
     assert(target_cpu == "x64")
-    ldflags = [ "/CETCOMPAT" ]
+    if (is_win) {
+      ldflags = [ "/CETCOMPAT" ]
+    } else if (is_linux) {
+      cflags = [ "-fcf-protection=full" ]
+      ldflags = [ "-fcf-protection=full", "-Wl,-z,shstk" ]
+      rustflags = [ "-Zcf-protection=full" ]
+    }
   }
 }
diff --git a/build/config/linux/BUILD.gn b/build/config/linux/BUILD.gn
index 131bb71d1d..b584e8d547 100644
--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -20,11 +20,13 @@ config("compiler") {
     cflags = []
     asmflags = []
     if (arm_control_flow_integrity == "standard") {
-      cflags += [ "-mbranch-protection=standard" ]
-      asmflags += [ "-mbranch-protection=standard" ]
+      cflags += [ "-mbranch-protection=pac-ret+leaf+bti" ]
+      asmflags += [ "-mbranch-protection=pac-ret+leaf+bti" ]
+      rustflags += [ "-Zbranch-protection=bti,pac-ret,leaf" ]
     } else if (arm_control_flow_integrity == "pac") {
-      cflags += [ "-mbranch-protection=pac-ret" ]
-      asmflags += [ "-mbranch-protection=pac-ret" ]
+      cflags += [ "-mbranch-protection=pac-ret+leaf" ]
+      asmflags += [ "-mbranch-protection=pac-ret+leaf" ]
+      rustflags += [ "-Zbranch-protection=pac-ret,leaf" ]
     }
   }
 }
diff --git a/build/config/sanitizers/BUILD.gn b/build/config/sanitizers/BUILD.gn
index 66b993426a..987e6c80f9 100644
--- a/build/config/sanitizers/BUILD.gn
+++ b/build/config/sanitizers/BUILD.gn
@@ -374,10 +374,10 @@ config("cfi_flags") {
 
     if (use_cfi_icall) {
       cflags += [ "-fsanitize=cfi-icall" ]
-      # TODO(crbug.com/40266913): Add cflags += [
-      # "-fsanitize-cfi-icall-experimental-normalize-integers" ]
-      # TODO(crbug.com/40266913): Add rustflags += [
-      # "-Zsanitizer-cfi-normalize-integers" ].
+      if (toolchain_supports_rust_thin_lto) {
+        cflags += [ "-fsanitize-cfi-icall-experimental-normalize-integers" ]
+        rustflags += [ "-Zsanitizer=cfi", "-Zsanitizer-cfi-normalize-integers" ]
+      }
     }
 
     if (use_cfi_diag) {
diff --git a/v8/BUILD.gn b/v8/BUILD.gn
index f2f72a02..b41fca74 100644
--- a/v8/BUILD.gn
+++ b/v8/BUILD.gn
@@ -821,7 +821,7 @@ assert(!v8_enable_snapshot_compression || v8_use_zlib,
        "Snapshot compression requires zlib")
 
 assert(!v8_enable_cet_shadow_stack ||
-           (v8_target_cpu == "x64" && target_os == "win"),
+           (v8_target_cpu == "x64" && (target_os == "win" || target_os == "linux")),
        "CET shadow stack is supported only on x64 Windows")
 
 if (v8_enable_sticky_mark_bits) {
