diff --git a/chrome/browser/sharing_hub/sharing_hub_features.cc b/chrome/browser/sharing_hub/sharing_hub_features.cc
index 2ddc318a2b738..6fd447d8e5ce1 100644
--- a/chrome/browser/sharing_hub/sharing_hub_features.cc
+++ b/chrome/browser/sharing_hub/sharing_hub_features.cc
@@ -72,7 +72,7 @@ BASE_FEATURE(kDesktopScreenshots,
 
 #if !BUILDFLAG(IS_ANDROID) && !BUILDFLAG(IS_CHROMEOS)
 void RegisterProfilePrefs(PrefRegistrySimple* registry) {
-  registry->RegisterBooleanPref(prefs::kDesktopSharingHubEnabled, true);
+  registry->RegisterBooleanPref(prefs::kDesktopSharingHubEnabled, false);
 }
 #endif
 
diff --git a/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_prefs.cc b/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_prefs.cc
index f792842f91922..28cc7122da39c 100644
--- a/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_prefs.cc
+++ b/chrome/browser/ui/toolbar/chrome_labs/chrome_labs_prefs.cc
@@ -49,7 +49,7 @@ const int kChromeLabsActivationThresholdDefaultValue = -1;
 const int kChromeLabsNewExperimentPrefValue = -1;
 
 void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
-  registry->RegisterBooleanPref(kBrowserLabsEnabledEnterprisePolicy, true);
+  registry->RegisterBooleanPref(kBrowserLabsEnabledEnterprisePolicy, false);
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   registry->RegisterDictionaryPref(kChromeLabsNewBadgeDictAshChrome);
 #endif
diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
index 886af67983837..120272ceacfa8 100644
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -243,11 +243,7 @@ BASE_FEATURE(kTabGroupsCollapseFreezing,
 // https://crbug.com/928954
 BASE_FEATURE(kTabHoverCardImages,
              "TabHoverCardImages",
-#if BUILDFLAG(IS_MAC)
              base::FEATURE_DISABLED_BY_DEFAULT
-#else
-             base::FEATURE_ENABLED_BY_DEFAULT
-#endif
 );
 
 const char kTabHoverCardImagesNotReadyDelayParameterName[] =
diff --git a/components/history_clusters/core/features.cc b/components/history_clusters/core/features.cc
index 897479817c607..a5fad5e8bb7fd 100644
--- a/components/history_clusters/core/features.cc
+++ b/components/history_clusters/core/features.cc
@@ -26,7 +26,7 @@ constexpr auto enabled_by_default_desktop_only =
 
 namespace internal {
 
-BASE_FEATURE(kJourneys, "Journeys", enabled_by_default_desktop_only);
+BASE_FEATURE(kJourneys, "Journeys", base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kJourneysImages,
              "JourneysImages",
