diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
index 7613006656aaa..ee707f847ccba 100644
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -4343,6 +4343,48 @@ const FeatureEntry kFeatureEntries[] = {
 // //tools/flags/generate_unexpire_flags.py.
 #include "build/chromeos_buildflags.h"
 #include "chrome/browser/unexpire_flags_gen.inc"
+    {"middle-click-autoscroll", "Autoscroll with Middleclick",
+     "Enables scroll with middleclick. Disabled by default. This feature "
+     "is exposed by Trivalent.", kOsLinux,
+     FEATURE_VALUE_TYPE(blink::features::kMiddleClickAutoscroll)},
+    {"show-punycode-domains", "Show punycode for IDN domains",
+     "Shows punycode for IDN domains to mitigate IDN homograph attacks. "
+     "Disabled by default. This feature is provided by Trivalent.",
+     kOsAll, FEATURE_VALUE_TYPE(url::kShowPunycodeDomains)},
+    {"pdf-javascript", "Enable PDF JavaScript",
+     "Toggle JavaScript for rendered PDFs (this also disables XFA). "
+     "Enabled by default. This feature is provided by Trivalent.",
+     kOsAll, FEATURE_VALUE_TYPE(chrome_pdf::features::kPdfJavaScript)},
+    {"clear-cross-origin-referrers", "Clear cross-origin referrers",
+     "Clears referrers when navigating across origins. Disabled by default. "
+     "This feature is provided by Trivalent.", kOsAll,
+     FEATURE_VALUE_TYPE(net::features::kDisableCrossOriginReferrers)},
+    {"cross-origin-trim-referrer", "Cross-origin referrer trimming",
+     "Trims the referrer to just the origin on cross origin navigation. "
+     "Enabled by default. This feature is exposed by Trivalent.",
+     kOsAll,
+     FEATURE_VALUE_TYPE(net::features::kCapReferrerToOriginOnCrossOrigin)},
+    {"gtk-version-4", "GTK Version 4",
+     "Enables Trivalent to use GTK 4 instead of GTK 3, useful for IME. "
+     "This flag is exposed by Trivalent.", kOsAll,
+     SINGLE_VALUE_TYPE_AND_VALUE("gtk-version", "4")},
+    {"strict-popup-blocking", "Strict Popup Blocking",
+     "Controls the strictness of the popup blocker. This switch is provided "
+     "by Trivalent.", kOsAll,
+     SINGLE_DISABLE_VALUE_TYPE("disable-strict-popup-blocking")},
+    {"hide-profile-icon", "Hide profile icon in toolbar",
+     "Hides the profile icon in the toolbar in regular profiles. Enabled "
+     "by default. This feature is provided by Trivalent." , kOsAll,
+     FEATURE_VALUE_TYPE(features::kHideProfileIcon)},
+    {"internal-page-jit", "Internal Page Jit",
+     "Toggle JIT for JavaScript and WASM on internal pages & extensions. "
+     "Disabled by default. This feature is provided by Trivalent.",
+     kOsAll, FEATURE_VALUE_TYPE(features::kEnableInternalPageJit)},
+    {"gssapi-support", "GSSAPI Authentication",
+     "Enables GSSAPI for authentication. WARNING! This can cause the "
+     "network service sandbox to become persistently disabled, enable only "
+     "if absolutely necessary. This switch is provided by Trivalent.",
+     kOsLinux, SINGLE_VALUE_TYPE("enable-gssapi")},
     {variations::switches::kEnableBenchmarking,
      flag_descriptions::kEnableBenchmarkingName,
      flag_descriptions::kEnableBenchmarkingDescription, kOsAll,
