diff --git a/v8/src/wasm/interpreter/wasm-interpreter-runtime.cc b/v8/src/wasm/interpreter/wasm-interpreter-runtime.cc
index 401bf9c181a..4127f1d3c2d 100644
--- a/v8/src/wasm/interpreter/wasm-interpreter-runtime.cc
+++ b/v8/src/wasm/interpreter/wasm-interpreter-runtime.cc
@@ -2132,7 +2132,7 @@ int WasmInterpreterRuntime::instruction_table_offset() {
   return OFFSET_OF(WasmInterpreterRuntime, instruction_table_);
 }
 
-#ifndef V8_DRUMBRAKE_BOUNDS_CHECKS
+#if defined(V8_ENABLE_DRUMBRAKE_TRACING) && !defined(V8_DRUMBRAKE_BOUNDS_CHECKS)
 // static
 int WasmInterpreterRuntime::trace_pop_func_offset() {
   return OFFSET_OF(WasmInterpreterRuntime, trace_pop_func_);
@@ -2152,7 +2152,8 @@ int WasmInterpreterRuntime::trace_push_func_offset() {
 int WasmInterpreterRuntime::trace_replace_func_offset() {
   return OFFSET_OF(WasmInterpreterRuntime, trace_replace_func_);
 }
-#endif  // V8_DRUMBRAKE_BOUNDS_CHECKS
+#endif  // defined(V8_ENABLE_DRUMBRAKE_TRACING) &&
+        // !defined(V8_DRUMBRAKE_BOUNDS_CHECKS)
 
 struct StackHandlerMarker {
   Address next;
