diff --git a/components/content_settings/core/browser/content_settings_default_provider.cc b/components/content_settings/core/browser/content_settings_default_provider.cc
index 4d550b6b82..2f564ed6e3 100644
--- a/components/content_settings/core/browser/content_settings_default_provider.cc
+++ b/components/content_settings/core/browser/content_settings_default_provider.cc
@@ -91,6 +91,17 @@ const std::string& GetPrefName(ContentSettingsType type) {
 
 }  // namespace
 
+namespace {
+
+#if BUILDFLAG(IS_LINUX)
+constexpr char kDownstreamJavascriptJitSettingToClear[] =
+    "profile.default_content_setting_values.javascript_jit";
+constexpr char kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref[] =
+    "profile.did_work_around_downstream_javascript_jit_default";
+#endif  // BUILDFLAG(IS_ANDROID)
+
+}  // namespace
+
 // static
 void DefaultProvider::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
@@ -127,6 +138,10 @@ void DefaultProvider::RegisterProfilePrefs(
   // TODO(https://crbug.com/367181093): clean this up.
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
 #endif  // !BUILDFLAG(IS_IOS)
+
+#if BUILDFLAG(IS_LINUX)
+  registry->RegisterBooleanPref(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, false);
+#endif // BUILDFLAG(IS_LINUX)
 }
 
 DefaultProvider::DefaultProvider(PrefService* prefs,
@@ -368,6 +383,13 @@ void DefaultProvider::DiscardOrMigrateObsoletePreferences() {
   // TODO(https://crbug.com/367181093): clean this up.
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
 #endif  // !BUILDFLAG(IS_IOS)
+
+#if BUILDFLAG(IS_LINUX)
+  if (!prefs_->GetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref)) {
+    prefs_->ClearPref(kDownstreamJavascriptJitSettingToClear);
+    prefs_->SetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, true);
+  }
+#endif // BUILDFLAG(IS_LINUX)
 }
 
 void DefaultProvider::RecordHistogramMetrics() {
diff --git a/components/content_settings/core/browser/content_settings_pref_provider.cc b/components/content_settings/core/browser/content_settings_pref_provider.cc
index c7aeb13ca4..b1c56c6cba 100644
--- a/components/content_settings/core/browser/content_settings_pref_provider.cc
+++ b/components/content_settings/core/browser/content_settings_pref_provider.cc
@@ -69,6 +69,17 @@ constexpr char kBug364820109AlreadyWorkedAroundPref[] =
 
 }  // namespace
 
+namespace {
+
+#if BUILDFLAG(IS_LINUX)
+constexpr char kDownstreamJavascriptJitSettingToClear[] =
+    "profile.content_settings.exceptions.javascript_jit";
+constexpr char kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref[] =
+    "profile.did_work_around_downstream_javascript_jit_exceptions";
+#endif // BUILDFLAG(IS_LINUX)
+
+}  // namespace
+
 // ////////////////////////////////////////////////////////////////////////////
 // PrefProvider:
 //
@@ -107,6 +118,10 @@ void PrefProvider::RegisterProfilePrefs(
   // TODO(https://crbug.com/367181093): clean this up.
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
 #endif  // !BUILDFLAG(IS_IOS)
+
+#if BUILDFLAG(IS_LINUX)
+  registry->RegisterBooleanPref(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, false);
+#endif // BUILDFLAG(IS_LINUX)
 }
 
 PrefProvider::PrefProvider(PrefService* prefs,
@@ -446,6 +461,13 @@ void PrefProvider::DiscardOrMigrateObsoletePreferences() {
   // TODO(https://crbug.com/367181093): clean this up.
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
 #endif  // !BUILDFLAG(IS_IOS)
+
+#if BUILDFLAG(IS_LINUX)
+  if (!prefs_->GetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref)) {
+    prefs_->ClearPref(kDownstreamJavascriptJitSettingToClear);
+    prefs_->SetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, true);
+  }
+#endif // BUILDFLAG(IS_LINUX)
 }
 
 void PrefProvider::SetClockForTesting(const base::Clock* clock) {
