diff --git a/media/base/media_switches.cc b/media/base/media_switches.cc
index 4df38fe102c1b..1d52e585902c1 100644
--- a/media/base/media_switches.cc
+++ b/media/base/media_switches.cc
@@ -1657,6 +1657,11 @@ BASE_FEATURE(kUseWindowBoundsForPip,
              "UseWindowBoundsForPip",
              base::FEATURE_ENABLED_BY_DEFAULT);
 
+// Enables FFmpeg allow lists for supported codecs / containers.
+BASE_FEATURE(kFFmpegAllowLists,
+             "FFmpegAllowLists",
+	     base::FEATURE_DISABLED_BY_DEFAULT);
+
 #if BUILDFLAG(IS_WIN)
 // Enables audio offload when supported by endpoints.
 BASE_FEATURE(kAudioOffload, "AudioOffload", base::FEATURE_DISABLED_BY_DEFAULT);
diff --git a/media/base/media_switches.h b/media/base/media_switches.h
index a659f74765496..1dd65d9583459 100644
--- a/media/base/media_switches.h
+++ b/media/base/media_switches.h
@@ -515,6 +515,7 @@ MEDIA_EXPORT BASE_DECLARE_FEATURE(kFuchsiaMediacodecVideoEncoder);
 MEDIA_EXPORT BASE_DECLARE_FEATURE(kVideoDecodeBatching);
 
 MEDIA_EXPORT BASE_DECLARE_FEATURE(kUseWindowBoundsForPip);
+MEDIA_EXPORT BASE_DECLARE_FEATURE(kFFmpegAllowLists);
 
 MEDIA_EXPORT BASE_DECLARE_FEATURE(kMediaLogToConsole);
 
diff --git a/media/ffmpeg/ffmpeg_common.cc b/media/ffmpeg/ffmpeg_common.cc
index cfd0552ee3108..9e1dd96a1e77a 100644
--- a/media/ffmpeg/ffmpeg_common.cc
+++ b/media/ffmpeg/ffmpeg_common.cc
@@ -19,6 +19,7 @@
 #include "media/base/audio_decoder_config.h"
 #include "media/base/decoder_buffer.h"
 #include "media/base/encryption_scheme.h"
+#include "media/base/media_switches.h"
 #include "media/base/media_util.h"
 #include "media/base/supported_types.h"
 #include "media/base/video_aspect_ratio.h"
@@ -76,7 +77,7 @@ const char* GetAllowedVideoDecoders() {
 void ApplyCodecContextSecuritySettings(AVCodecContext* codec_context) {
   // Future versions of ffmpeg may copy the allow list from the format
   // context.
-  if (!codec_context->codec_whitelist) {
+  if (base::FeatureList::IsEnabled(kFFmpegAllowLists) && !codec_context->codec_whitelist) {
     // Note: FFmpeg will try to free this string, so we must duplicate it.
     codec_context->codec_whitelist =
         av_strdup(codec_context->codec_type == AVMEDIA_TYPE_AUDIO
diff --git a/media/filters/ffmpeg_glue.cc b/media/filters/ffmpeg_glue.cc
index 26bb9e8b92614..29f2e7c02747f 100644
--- a/media/filters/ffmpeg_glue.cc
+++ b/media/filters/ffmpeg_glue.cc
@@ -131,8 +131,10 @@ FFmpegGlue::FFmpegGlue(FFmpegURLProtocol* protocol) {
   // memory usage.
   //
   // Note: FFmpeg will try to free these strings, so we must duplicate them.
-  format_context_->codec_whitelist = av_strdup(GetAllowedAudioDecoders());
-  format_context_->format_whitelist = av_strdup(GetAllowedDemuxers());
+  if (base::FeatureList::IsEnabled(kFFmpegAllowLists)) {
+    format_context_->codec_whitelist = av_strdup(GetAllowedAudioDecoders());
+    format_context_->format_whitelist = av_strdup(GetAllowedDemuxers());
+  }
 }
 
 bool FFmpegGlue::OpenContext(bool is_local_file) {
