diff --git a/components/search_engines/search_engine_choice/search_engine_choice_service.cc b/components/search_engines/search_engine_choice/search_engine_choice_service.cc
index c152260117ccc..904423cc3095b 100644
--- a/components/search_engines/search_engine_choice/search_engine_choice_service.cc
+++ b/components/search_engines/search_engine_choice/search_engine_choice_service.cc
@@ -256,13 +256,6 @@ SearchEngineChoiceService::GetStaticChoiceScreenConditions(
     return SearchEngineChoiceScreenConditions::kAlreadyCompleted;
   }
 
-  int country_id = GetCountryId();
-  DVLOG(1) << "Checking country for choice screen, found: "
-           << country_codes::CountryIDToCountryString(country_id);
-  if (!IsEeaChoiceCountry(country_id)) {
-    return SearchEngineChoiceScreenConditions::kNotInRegionalScope;
-  }
-
   // Initially exclude users with this type of override. Consult b/302675777 for
   // next steps.
   if (profile_prefs_->HasPrefPath(prefs::kSearchProviderOverrides)) {
diff --git a/chrome/browser/search_engine_choice/search_engine_choice_dialog_service_factory.cc b/chrome/browser/search_engine_choice/search_engine_choice_dialog_service_factory.cc
index 96286eaf96db2..75f0eee1d82c4 100644
--- a/chrome/browser/search_engine_choice/search_engine_choice_dialog_service_factory.cc
+++ b/chrome/browser/search_engine_choice/search_engine_choice_dialog_service_factory.cc
@@ -128,10 +128,6 @@ SearchEngineChoiceDialogServiceFactory::BuildServiceInstanceForBrowserContext(
 
   base::CommandLine* const command_line =
       base::CommandLine::ForCurrentProcess();
-  if (!g_is_chrome_build &&
-      !command_line->HasSwitch(switches::kForceSearchEngineChoiceScreen)) {
-    return nullptr;
-  }
 
   if (command_line->HasSwitch(switches::kNoFirstRun) &&
       !command_line->HasSwitch(
