diff --git a/ui/base/ui_base_features.cc b/ui/base/ui_base_features.cc
index fbaea143b6..3184d5bc05 100644
--- a/ui/base/ui_base_features.cc
+++ b/ui/base/ui_base_features.cc
@@ -149,7 +149,7 @@ BASE_FEATURE(kWaylandSessionManagement,
 COMPONENT_EXPORT(UI_BASE_FEATURES)
 BASE_FEATURE(kOverrideDefaultOzonePlatformHintToAuto,
              "OverrideDefaultOzonePlatformHintToAuto",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 #endif  // BUILDFLAG(IS_LINUX)
 
 // Chrome for Linux should eventually use XInput2 key events.
