diff --git a/components/omnibox/browser/autocomplete_controller.cc b/components/omnibox/browser/autocomplete_controller.cc
index 22af8003c7..54c42a6871 100644
--- a/components/omnibox/browser/autocomplete_controller.cc
+++ b/components/omnibox/browser/autocomplete_controller.cc
@@ -22,6 +22,7 @@
 #include <utility>
 
 #include "base/check_op.h"
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/format_macros.h"
 #include "base/functional/bind.h"
@@ -549,6 +550,19 @@ AutocompleteController::AutocompleteController(
           metrics::OmniboxEventProto::UNKNOWN_POSITION),
       config_(config) {
   config_.provider_types &= ~OmniboxFieldTrial::GetDisabledProviderTypes();
+  const std::string kOmniboxScopeFlag = "scope-omnibox-autocomplete";
+  const base::CommandLine* cmdline = base::CommandLine::ForCurrentProcess();
+  if (cmdline->HasSwitch(kOmniboxScopeFlag)) {
+    std::vector<std::string> switch_values = base::SplitString(
+        cmdline->GetSwitchValueASCII(kOmniboxScopeFlag), ",",
+        base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+    config_.provider_types &= AutocompleteProvider::TYPE_KEYWORD | AutocompleteProvider::TYPE_SEARCH |
+        AutocompleteProvider::TYPE_HISTORY_URL;
+    if (std::ranges::contains(switch_values, "bookmarks"))
+      config_.provider_types |= AutocompleteProvider::TYPE_BOOKMARK;
+    if (std::ranges::contains(switch_values, "chrome"))
+      config_.provider_types |= AutocompleteProvider::TYPE_BUILTIN;
+  }
 
   // Providers run in the order they're added. Async providers should run first
   // so their async requests can be kicked off before waiting a few milliseconds
diff --git a/components/omnibox/browser/history_url_provider.cc b/components/omnibox/browser/history_url_provider.cc
index 526c06fdfe..16135f689c 100644
--- a/components/omnibox/browser/history_url_provider.cc
+++ b/components/omnibox/browser/history_url_provider.cc
@@ -487,6 +487,9 @@ void HistoryURLProvider::Start(const AutocompleteInput& input,
   if (fixed_up_input.type() != metrics::OmniboxInputType::QUERY)
     matches_.push_back(what_you_typed_match);
 
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("scope-omnibox-autocomplete"))
+    return;
+
   // We'll need the history service to run both passes, so try to obtain it.
   history::HistoryService* const history_service =
       client()->GetHistoryService();
diff --git a/components/omnibox/browser/search_provider.cc b/components/omnibox/browser/search_provider.cc
index ff56a4fb37..bea9159cde 100644
--- a/components/omnibox/browser/search_provider.cc
+++ b/components/omnibox/browser/search_provider.cc
@@ -15,6 +15,7 @@
 #include <vector>
 
 #include "base/base64.h"
+#include "base/command_line.h"
 #include "base/feature_list.h"
 #include "base/functional/bind.h"
 #include "base/functional/callback.h"
@@ -682,6 +683,9 @@ void SearchProvider::Run(bool query_is_private) {
 }
 
 void SearchProvider::DoHistoryQuery(bool minimal_changes) {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("scope-omnibox-autocomplete"))
+    return;
+
   // The history query results are synchronous, so if minimal_changes is true,
   // we still have the last results and don't need to do anything.
   if (minimal_changes)
diff --git a/components/url_formatter/url_fixer.cc b/components/url_formatter/url_fixer.cc
index d5d9a14c8d..6875acd917 100644
--- a/components/url_formatter/url_fixer.cc
+++ b/components/url_formatter/url_fixer.cc
@@ -10,6 +10,8 @@
 #include <string_view>
 
 #include "base/check_op.h"
+#include "base/command_line.h"
+#include "base/strings/string_split.h"
 #include "base/compiler_specific.h"
 #include "base/files/file_path.h"
 #include "base/files/file_util.h"
@@ -635,7 +637,16 @@ GURL FixupURLInternal(const std::string& text,
                  desired_tld, &url);
     }
     if (chrome_url && !parts.host.is_valid()) {
-      url.append(kChromeUIDefaultHost);
+      bool append = true;
+      const std::string kOmniboxScopeFlag = "scope-omnibox-autocomplete";
+      const base::CommandLine* cmdline = base::CommandLine::ForCurrentProcess();
+      if (cmdline->HasSwitch(kOmniboxScopeFlag)) {
+        std::vector<std::string> switch_values = base::SplitString(
+            cmdline->GetSwitchValueASCII(kOmniboxScopeFlag), ",",
+            base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+        append = std::ranges::contains(switch_values, "chrome");
+      }
+      if (append) url.append(kChromeUIDefaultHost);
     }
     FixupPort(trimmed, parts.port, &url);
     FixupPath(trimmed, parts.path, &url);
