diff --git a/chrome/app/settings_strings.grdp b/chrome/app/settings_strings.grdp
index 81776c22d5..88e6f8d392 100644
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1322,6 +1322,46 @@
     Browsing and searching is faster than standard preloading
   </message>

+  <!-- Trivalent -->
+  <message name="IDS_WEBSITE_DARK_MODE" desc="Trivalent String.">
+    Website dark mode
+  </message>
+  <message name="IDS_PRINTING_CAPABILITIES" desc="Trivalent String.">
+    Printing capabilities
+  </message>
+
+  <!-- Trivalent security section -->
+  <message name="IDS_HARDENING_TITLE" desc="Trivalent String.">
+    Hardening
+  </message>
+  <message name="IDS_INCOGNITO_LAUNCH_LABEL" desc="Trivalent String.">
+    Incognito Launch
+  </message>
+  <message name="IDS_INCOGNITO_LAUNCH_DESCRIPTION" desc="Trivalent String.">
+    Opens new sessions and external links in Incognito tabs
+  </message>
+  <message name="IDS_DISABLE_3D_APIS_LABEL" desc="Trivalent String.">
+    Disable 3D APIs
+  </message>
+  <message name="IDS_DISABLE_3D_API_DESCRIPTION" desc="Trivalent String.">
+    Disable web-facing access to features like WebGL
+  </message>
+  <message name="IDS_WEBRTC_HANDLING_POLICY_LABEL" desc="Trivalent String.">
+    WebRTC Handling Policy
+  </message>
+  <message name="IDS_NETWORK_SERVICE_SANDBOX_LABEL" desc="Trivalent String.">
+    Network Service Sandbox
+  </message>
+  <message name="IDS_NETWORK_SERVICE_SANDBOX_DESCRIPTION" desc="Trivalent String.">
+    Disabling may fix issues with data persistence but will reduce security slightly, requires a restart to apply
+  </message>
+  <message name="IDS_DISABLE_EXTENSIONS_LABEL" desc="Trivalent String.">
+    Disable Extensions
+  </message>
+  <message name="IDS_NETWORK_SERVICE_SANDBOX_DESCRIPTION" desc="Trivalent String.">
+    Changes to this setting requires a restart to apply
+  </message>
+
   <!-- Privacy Page -->
   <message name="IDS_SETTINGS_PRIVACY" desc="Name of the settings page which allows users to modify privacy and security settings.">
     Privacy and security
