diff --git a/chrome/app/settings_strings.grdp b/chrome/app/settings_strings.grdp
index 81776c22d5..88e6f8d392 100644
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1322,6 +1322,46 @@
     Browsing and searching is faster than standard preloading
   </message>

+  <!-- Trivalent -->
+  <message name="IDS_WEBSITE_DARK_MODE" desc="Trivalent String.">
+    Website dark mode
+  </message>
+  <message name="IDS_PRINTING_CAPABILITIES" desc="Trivalent String.">
+    Printing capabilities
+  </message>
+
+  <!-- Trivalent security section -->
+  <message name="IDS_HARDENING_TITLE" desc="Trivalent String.">
+    Hardening
+  </message>
+  <message name="IDS_INCOGNITO_LAUNCH_LABEL" desc="Trivalent String.">
+    Incognito Launch
+  </message>
+  <message name="IDS_INCOGNITO_LAUNCH_DESCRIPTION" desc="Trivalent String.">
+    Opens new sessions and external links in Incognito tabs
+  </message>
+  <message name="IDS_DISABLE_3D_APIS_LABEL" desc="Trivalent String.">
+    Disable 3D APIs
+  </message>
+  <message name="IDS_DISABLE_3D_APIS_DESCRIPTION" desc="Trivalent String.">
+    Disable web-facing access to features like WebGL
+  </message>
+  <message name="IDS_WEBRTC_HANDLING_POLICY_LABEL" desc="Trivalent String.">
+    WebRTC Handling Policy
+  </message>
+  <message name="IDS_NETWORK_SERVICE_SANDBOX_LABEL" desc="Trivalent String.">
+    Network Service Sandbox
+  </message>
+  <message name="IDS_NETWORK_SERVICE_SANDBOX_DESCRIPTION" desc="Trivalent String.">
+    Disabling may fix issues with data persistence but will reduce security slightly, requires a restart to apply
+  </message>
+  <message name="IDS_DISABLE_EXTENSIONS_LABEL" desc="Trivalent String.">
+    Disable Extensions
+  </message>
+  <message name="IDS_DISABLE_EXTENSIONS_DESCRIPTION" desc="Trivalent String.">
+    Changes to this setting requires a restart to apply
+  </message>
+
   <!-- Privacy Page -->
   <message name="IDS_SETTINGS_PRIVACY" desc="Name of the settings page which allows users to modify privacy and security settings.">
     Privacy and security
diff --git a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
index 556a396168..a6547ce454 100644
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -510,6 +510,7 @@ void AddAiStrings(content::WebUIDataSource* html_source) {
 void AddAppearanceStrings(content::WebUIDataSource* html_source,
                           Profile* profile) {
   static constexpr webui::LocalizedString kLocalizedStrings[] = {
+      {"websiteDarkMode", IDS_WEBSITE_DARK_MODE},
       {"appearancePageTitle", IDS_SETTINGS_APPEARANCE},
       {"customWebAddress", IDS_SETTINGS_CUSTOM_WEB_ADDRESS},
       {"enterCustomWebAddress", IDS_SETTINGS_ENTER_CUSTOM_WEB_ADDRESS},
@@ -2129,6 +2130,16 @@ bool ShouldLinkSecureDnsOsSettings() {
 void AddPrivacyStrings(content::WebUIDataSource* html_source,
                        Profile* profile) {
   static constexpr webui::LocalizedString kLocalizedStrings[] = {
+      {"hardeningSectionTitle", IDS_HARDENING_TITLE},
+      {"incognitoLaunchLabel", IDS_INCOGNITO_LAUNCH_LABEL},
+      {"incognitoLaunchDesc", IDS_INCOGNITO_LAUNCH_DESCRIPTION},
+      {"disable3dApisLabel", IDS_DISABLE_3D_APIS_LABEL},
+      {"disable3dApisDesc", IDS_DISABLE_3D_APIS_DESCRIPTION},
+      {"webrtcHandlingPolicyLabel", IDS_WEBRTC_HANDLING_POLICY_LABEL},
+      {"netServSandboxLabel", IDS_NETWORK_SERVICE_SANDBOX_LABEL},
+      {"netServSandboxDesc", IDS_NETWORK_SERVICE_SANDBOX_DESCRIPTION},
+      {"disableExtensionsLabel", IDS_DISABLE_EXTENSIONS_LABEL},
+      {"disableExtensionsDesc", IDS_DISABLE_EXTENSIONS_DESCRIPTION},
       {"privacyPageTitle", IDS_SETTINGS_PRIVACY},
       {"privacyPageMore", IDS_SETTINGS_PRIVACY_MORE},
       {"doNotTrack", IDS_SETTINGS_ENABLE_DO_NOT_TRACK},
@@ -3826,6 +3837,7 @@ void AddSiteDataPageStrings(content::WebUIDataSource* html_source,
 #if !BUILDFLAG(IS_CHROMEOS)
 void AddSystemStrings(content::WebUIDataSource* html_source) {
   static constexpr webui::LocalizedString kLocalizedStrings[] = {
+      {"printingCapabilities", IDS_PRINTING_CAPABILITIES},
       {"systemPageTitle", IDS_SETTINGS_SYSTEM},
 #if !BUILDFLAG(IS_MAC)
       {"backgroundAppsLabel", IDS_SETTINGS_SYSTEM_BACKGROUND_APPS_LABEL},
